zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/DocumentControllerTest.java

```diff
@@ -187,6 +274,253 @@ void shouldCreateDocumentWithMetadata() {
+
+    final var om = new ObjectMapper();
+    final var multipartBodyBuilder = new MultipartBodyBuilder();
+    multipartBodyBuilder
+        .part("files", content1)
+        .header(
+            "X-Document-Metadata",
+            om.writeValueAsString(
+                new DocumentDetails()
+                    .contentType(contentType.toString())
+                    .fileName(filename1)
+                    .expiresAt(timestamp.toString())
+                    .processInstanceKey(123L)));
+    multipartBodyBuilder
+        .part("files", content2)
+        .header(
+            "X-Document-Metadata",
+            om.writeValueAsString(
+                new DocumentDetails()
+                    .contentType(contentType.toString())
+                    .fileName(filename2)
+                    .expiresAt(timestamp.toString())
+                    .processInstanceKey(123L)));
+
+    // when/then
+    webClient
+        .post()
+        .uri(DOCUMENTS_BASE_URL + "/batch")
+        .contentType(MediaType.MULTIPART_FORM_DATA)
+        .bodyValue(multipartBodyBuilder.build())
+        .accept(RequestMapper.MEDIA_TYPE_KEYS_NUMBER)
+        .exchange()
+        .expectStatus()
+        .isCreated()
+        .expectBody()
+        .json(
+            String.format(
+                """
+                    {
+                      "createdDocuments": [
+                        {
+                          "metadata": {
+                            "processInstanceKey": 123,
+                            "contentType": "application/octet-stream",
+                            "fileName": "file.txt",
+                            "expiresAt": "%s",
+                            "size": 0,
+                            "customProperties": {}
+                          },
+                          "camunda.document.type": "camunda",
+                          "storeId": "default",
+                          "documentId": "documentId",
+                          "contentHash": "dummy_hash"
+                        },
+                        {
+                          "metadata": {
+                            "processInstanceKey": 123,
+                            "contentType": "application/octet-stream",
+                            "fileName": "file.txt",
+                            "expiresAt": "%s",
+                            "size": 0,
+                            "customProperties": {}
+                          },
+                          "camunda.document.type": "camunda",
+                          "storeId": "default",
+                          "documentId": "documentId",
+                          "contentHash": "dummy_hash"
+                        }
+                      ],
+                      "failedDocuments": []
+                    }
+                    """
+                    .formatted(timestamp, timestamp),
+                timestamp));
```

Comment:
🙃 Is there one unnecessary formatting going on here? 😅 We're formatting the string twice, I think.

Reply:
Yeah, I'll fix it. 😄